test/reporters/xunit.spec.js

```diff
@@ -339,6 +341,7 @@ describe('XUnit reporter', function() {
         }
       };
       runner.once = runner.on;
+      runner.stats = createStatsCollector(runner);
```

Comment:
Leave this to your call as partially outside your scope (don't remember "xunit.spec.js" being in previous commits).

The `runner` variable at L328 masks the one at L14. Would it be better to remove it and change L337-L344 to this?
```js
runner.on = runner.once = function(eventName, eventHandler) {
  // capture the event handler
  events[eventName] = eventHandler;
};
```

Reply:
Yes I agree its better to define `runner` (and attach stats) once and just re-use. I need to keep an eye out for that in these tests.
Have now made that change.

BTW yep the original commit didn't have this change, it came from the ownership changing from Reporter to Runner.

Reply:
> BTW yep the original commit didn't have this change, it came from the ownership changing from Reporter to Runner.

Any idea why? Both Base and Runner had a reference to the same "object". Who created it shouldn't have mattered, unless I'm really missing something.

Reply:
The Runner in this test is a mocked object. So the Reporter + Base require the mock to have a "stats" property they can attach to.
Does that make sense? I hope I haven't misunderstood what you meant.

Reply:
Hmm did you mean does it need the assignment? I think your right + no assignment is needed.